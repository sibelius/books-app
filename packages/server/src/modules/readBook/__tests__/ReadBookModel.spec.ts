import { sanitizeTestObject } from '@booksapp/test-utils';

import {
  clearDbAndRestartCounters,
  connectMongoose,
  disconnectMongoose,
  createUser,
  createReadBook,
  createBook,
} from '../../../../test/helpers';

import ReadBookModel, { IReadBook } from '../ReadBookModel';

beforeAll(connectMongoose);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

describe('ReadBookModel', () => {
  it('should create new read book using ReadBookModel', async () => {
    const user = await createUser();
    const book = await createBook({ pages: 10 });
    const readPages = 5;

    const readBook = await new ReadBookModel({
      userId: user._id,
      bookId: book._id,
      readPages,
    }).save();

    const readBookObj = await ReadBookModel.findOne({ _id: readBook._id }).lean<IReadBook>();

    expect(readBookObj?.userId).toMatchObject(user._id);
    expect(readBookObj?.bookId).toMatchObject(book._id);
    expect(readBookObj?.readPages).toBe(readPages);
    expect(sanitizeTestObject(readBookObj)).toMatchSnapshot();
  });

  it('should create new read book using createRow', async () => {
    const user = await createUser();
    const book = await createBook({ pages: 10 });

    const readBook = await createReadBook();

    const readBookObj = await ReadBookModel.findOne({ _id: readBook._id }).lean<IReadBook>();

    expect(readBookObj?.userId).toMatchObject(user._id);
    expect(readBookObj?.bookId).toMatchObject(book._id);
    expect(readBookObj?.readPages).toBe(1);
  });

  it('should create new read book with custom properties', async () => {
    const user = await createUser();
    const book = await createBook({ pages: 10 });
    const readPages = 5;

    const readBook = await createReadBook({ userId: user._id, bookId: book._id, readPages });

    const readBookObj = await ReadBookModel.findOne({ _id: readBook._id }).lean<IReadBook>();

    expect(readBookObj?.userId).toMatchObject(user._id);
    expect(readBookObj?.bookId).toMatchObject(book._id);
    expect(readBookObj?.readPages).toBe(readPages);
  });
});
