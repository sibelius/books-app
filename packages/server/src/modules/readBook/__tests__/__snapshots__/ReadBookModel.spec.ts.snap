// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadBookModel should create new read book using ReadBookModel 1`] = `
Object {
  "__v": 0,
  "_id": "ObjectId",
  "bookId": "ObjectId",
  "createdAt": "FROZEN-CREATEDAT",
  "isActive": true,
  "readPages": 5,
  "removedAt": "EMPTY",
  "updatedAt": "FROZEN-UPDATEDAT",
  "userId": "ObjectId",
}
`;
